src/Preview.js

```diff
@@ -31,6 +32,10 @@ export default class Preview extends Component {
       this.setState({ image: light })
       return
     }
+    if (typeof light !== "string" && typeof light !== "boolean") {
+      this.setState({ previewImage: { isComponent: true, component: light } });
+      return;
+    }
```

Comment:
I'm not sure it's a good idea to assume than if it isn't a string or boolean then it is definitely a component.  Maybe you could try 
```
 else if (React.isValidElement(light) {
        this.setState({ previewImage: { isComponent: true, component: light } });
	return;
}
```